armclang support broken

**Notes (delete this)**
Merging of PR: #60031 broke armclang support.

**Describe the bug**
PR #60031 has remove build targets from the build system, causing failures when building with armclang.
```
CMake Error at /projects/github/ncs/zephyr/cmake/linker/armlink/target.cmake:74 (target_link_libraries):
  Error evaluating generator expression:                                                                                                                                                                                                      

    $<TARGET_OBJECTS:arch__arm__core__aarch32__cortex_m>                                                                                                                                                                                      

  Objects of target "arch__arm__core__aarch32__cortex_m" referenced but no                                                                                                                                                                    
  such target exists.                                                                                                                                                                                                                         
Call Stack (most recent call first):                                                                                                                                                                                                          
  /projects/github/ncs/zephyr/CMakeLists.txt:1427 (toolchain_ld_link_elf)                                                                                                                                                                     

....
```

**To Reproduce**
Set ZEPHYR_TOOLCHAIN_VARIANT=armclang and build.

**Expected behavior**


**Impact**
Not possible to build using armclang

**Logs and console output**
See above.

**Environment (please complete the following information):**
 - Toolchain : armclang
 - PR #60031

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/cmake/linker/armlink/target.cmake
+++ b/cmake/linker/armlink/target.cmake
@@ -65,7 +65,7 @@ function(toolchain_ld_link_elf)
   )
 
   foreach(lib ${ZEPHYR_LIBS_PROPERTY})
-    if(NOT ${lib} STREQUAL arch__arm__core__aarch32__cortex_m)
+    if(NOT ${lib} STREQUAL arch__arm__core__cortex_m)
       list(APPEND ZEPHYR_LIBS_OBJECTS $<TARGET_OBJECTS:${lib}>)
       list(APPEND ZEPHYR_LIBS_OBJECTS $<TARGET_PROPERTY:${lib},LINK_LIBRARIES>)
     endif()
@@ -76,7 +76,7 @@ function(toolchain_ld_link_elf)
     ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_PRE_SCRIPT}
     --scatter=${TOOLCHAIN_LD_LINK_ELF_LINKER_SCRIPT}
     ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_POST_SCRIPT}
-    $<TARGET_OBJECTS:arch__arm__core__aarch32__cortex_m>
+    $<TARGET_OBJECTS:arch__arm__core__cortex_m>
     --map --list=${TOOLCHAIN_LD_LINK_ELF_OUTPUT_MAP}
     ${ZEPHYR_LIBS_OBJECTS}
     kernel

